cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -378,7 +378,7 @@ func (g *gitHubRepository) downloadFilesFromRelease(release *github.RepositoryRe
 	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
 		var redirect string
 		var downloadReleaseError error
-		reader, redirect, downloadReleaseError = client.Repositories.DownloadReleaseAsset(context.TODO(), g.owner, g.repository, *assetID, http.DefaultClient)
+		reader, redirect, downloadReleaseError = client.Repositories.DownloadReleaseAsset(ctx, g.owner, g.repository, *assetID, nil)
```

Comment:
Sorry not sure why I just saw that diff now. That's the go doc of the func
```go
// followRedirectsClient can be passed to download the asset from a redirected
// location. Passing http.DefaultClient is recommended unless special circumstances
// exist, but it's possible to pass any http.Client. If nil is passed the
// redirectURL will be returned instead.
```

This sounds like passing nil means that DownloadReleaseAsset doesn't follow redirects.

What am I missing?

Reply:
Was that just changed with the last force push?

Reply:
Okay tests are failing locally, guess I'm not missing anything :)

Reply:
Forgot to revert `nil` when testing the error handling 😂